packages/ec-core/src/generator/wrappers/owned.rs

```diff
@@ -0,0 +1,57 @@
+use std::{borrow::Borrow, marker::PhantomData, num::NonZeroUsize};
+
+use rand::{distributions::Uniform, prelude::Distribution};
+
+use crate::generator::wrappers::slice_cloning::EmptySlice;
+
+#[derive(Debug, PartialEq)]
+pub struct OneOfCloning<T, U> {
+    collection: T,
+    range: Uniform<usize>,
+    _p: PhantomData<U>,
+}
+
+impl<T, U> OneOfCloning<T, U>
+where
+    T: Borrow<[U]>,
```

Comment:
Just so I'm understanding, this says `T` can be borrowed as a slice of `U`, right? So we can pass in things like `Vec<U>` or array of `U` for `T`.

In a lot of other cases we've done things like say that we have to be able to convert `T` into an iterator over `U`, but here we don't want to do this because we need to store the collection in the instance of `OneOfCloning` and we don't want to have to iterate over `T` and make a new collection to store. Correct?

Reply:
yes exactly